### PR TITLE
Feature: Rework assets filter

### PR DESCRIPTION
**Pull request by borisermakof**

### CODE REVIEW

File: model/src/main/kotlin/org/digma/intellij/plugin/persistence/PersistenceData.kt [hunk 1]
```diff
@@ -1,5 +1,7 @@
 package org.digma.intellij.plugin.persistence
 
+import com.fasterxml.jackson.databind.JsonNode
```
Comment (shalom938) [inline model/src/main/kotlin/org/digma/intellij/plugin/persistence/PersistenceData.kt:3]: organize imports

File: src/main/java/org/digma/intellij/plugin/assets/AssetsIndexTemplateBuilder.java [hunk 2]
```diff
@@ -53,6 +54,7 @@ public InputStream build(Project project) {
             var data = new HashMap<String, Object>();
             JCefTemplateUtils.addCommonEnvVariables(data);
 
+            var Curenv = PersistenceService.getInstance().getState().getCurrentEnv();
```
Comment (shalom938) [inline src/main/java/org/digma/intellij/plugin/assets/AssetsIndexTemplateBuilder.java:57]: Please fix to java code stype, variable name starts with lower case.
Please ask Arik how to use our shared code style.

File: src/main/java/org/digma/intellij/plugin/assets/AssetsMessageRouterHandler.java [hunk 3]
```diff
@@ -4,30 +4,40 @@
...
 import org.digma.intellij.plugin.ui.settings.Theme;
 import org.jetbrains.annotations.NotNull;
 
+import java.util.HashMap;
+import java.util.Map;
+
 import static org.digma.intellij.plugin.common.StopWatchUtilsKt.stopWatchStart;
 import static org.digma.intellij.plugin.common.StopWatchUtilsKt.stopWatchStop;
+import static org.digma.intellij.plugin.ui.jcef.JCefBrowserUtilsKt.executeWindowPostMessageJavaScript;
+import static org.digma.intellij.plugin.ui.jcef.JCefBrowserUtilsKt.serializeAndExecuteWindowPostMessageJavaScript;
```
Comment (shalom938) [inline src/main/java/org/digma/intellij/plugin/assets/AssetsMessageRouterHandler.java:40]: organize imports

File: src/main/java/org/digma/intellij/plugin/assets/model/outgoing/SetCategoriesDataMessage.java [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.annotation.JsonSerialize;
+
+@JsonIgnoreProperties(ignoreUnknown = true)
+@JsonInclude(JsonInclude.Include.NON_NULL)
+@JsonAutoDetect(fieldVisibility = JsonAutoDetect.Visibility.ANY)
+@JsonSerialize
+public record SetCategoriesDataMessage(@JsonProperty(value = "type",required = true) String type,
+                                       @JsonProperty(value = "action",required = true)  String action,
```
Comment (shalom938) [inline src/main/java/org/digma/intellij/plugin/assets/model/outgoing/SetCategoriesDataMessage.java:16]: class not used

**Review by shalom938 — APPROVED** [review]

File: src/main/java/org/digma/intellij/plugin/assets/model/outgoing/SetCategoriesDataMessage.java (hunk 4 above)
Comment (borisermakof) [reply src/main/java/org/digma/intellij/plugin/assets/model/outgoing/SetCategoriesDataMessage.java:16]: fixed

File: src/main/java/org/digma/intellij/plugin/assets/AssetsMessageRouterHandler.java (hunk 3 above)
Comment (borisermakof) [reply src/main/java/org/digma/intellij/plugin/assets/AssetsMessageRouterHandler.java:40]: done

File: model/src/main/kotlin/org/digma/intellij/plugin/persistence/PersistenceData.kt (hunk 1 above)
Comment (borisermakof) [reply model/src/main/kotlin/org/digma/intellij/plugin/persistence/PersistenceData.kt:3]: done

File: src/main/java/org/digma/intellij/plugin/assets/AssetsIndexTemplateBuilder.java (hunk 2 above)
Comment (borisermakof) [reply src/main/java/org/digma/intellij/plugin/assets/AssetsIndexTemplateBuilder.java:57]: fixed

**Pull request merged**